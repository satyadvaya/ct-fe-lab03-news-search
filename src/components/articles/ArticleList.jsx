import React, { Component } from 'react';
import Article from './Article';
import PropTypes from 'prop-types';

export default function ArticleList({ articles }) {
  const ArticleList = articles.map(({ title, author, description }) => (
    <li key={`${title}-${author}-${description}`}>
      <Article title={title} author={author} description={description} />
    </li>
  ));
  return <ul aria-label="articles">{ArticleList}</ul>;
}

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string,
      description: PropTypes.string.isRequired,
    })
  ),
};
