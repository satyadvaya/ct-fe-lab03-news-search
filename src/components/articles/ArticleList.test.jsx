import React, { Component } from 'react';
import ArticleList from './ArticleList';
import { render, screen } from '@testing-library/react';

const testArticles = [
  {
    author: 'Dave Sebastian',
    title: 'Hertz Orders 100k Teslas',
    description:
      'Car-rental firm, recently emerged from bankruptcy under new ownership, orders vehicles by the end of 2022',
  },
  {
    author: 'Richard Rubin',
    title: 'LEVIES ON UNSOLD ASSETS',
    description:
      'Sen. Ron Wyden releases details of plan to impose multibillion-dollar tax bills on people such as Amazon founder Jeff Bezos and Tesla CEO Elon Musk',
  },
  {
    author: 'James Willhite',
    title:
      'Tesla, Merck, Moderna: What to Watch When the Stock Market Opens Today',
    description:
      'Tesla reported record deliveries in the third quarter; Merck shares were extending gains from last week',
  },
];

describe('Shows a list of articles', () => {
  it('should display a list of articles containing an input keyword', () => {
    render(<ArticleList articles={testArticles} />);

    const articleList = screen.getByRole('list', {
      name: 'articles',
    });

    expect(articleList).toMatchSnapshot();
  });
});
