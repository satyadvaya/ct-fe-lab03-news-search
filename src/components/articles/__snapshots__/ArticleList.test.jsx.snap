// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows a list of articles should display a list of articles containing an input keyword 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <h3>
      Hertz Orders 100k Teslas
    </h3>
    <p>
      Dave Sebastian
    </p>
    <p>
      Car-rental firm, recently emerged from bankruptcy under new ownership, orders vehicles by the end of 2022
    </p>
  </li>
  <li>
    <h3>
      LEVIES ON UNSOLD ASSETS
    </h3>
    <p>
      Richard Rubin
    </p>
    <p>
      Sen. Ron Wyden releases details of plan to impose multibillion-dollar tax bills on people such as Amazon founder Jeff Bezos and Tesla CEO Elon Musk
    </p>
  </li>
  <li>
    <h3>
      Tesla, Merck, Moderna: What to Watch When the Stock Market Opens Today
    </h3>
    <p>
      James Willhite
    </p>
    <p>
      Tesla reported record deliveries in the third quarter; Merck shares were extending gains from last week
    </p>
  </li>
</ul>
`;
